src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -207,7 +214,169 @@ void LinearIR::LoopManager::mark_loop(LinearIR::constExprIt loop_begin_pos,
+    for (const auto& exit_point : exit_points) {
+        const auto consumers_inputs = exit_point.expr_port->get_connected_ports();
+
+        std::set<LinearIR::LoopManager::LoopPort> mapped_entry_points;
+        std::set<ExpressionPtr> outside_consumers;
+        for (const auto& consumer_input : consumers_inputs) {
+            const auto entry_point_it = std::find_if(entry_points.begin(), entry_points.end(),
+                                                     [&consumer_input](const LoopManager::LoopPort& point) {
+                                                             return *point.expr_port.get() == consumer_input;
+                                                         });
+            if (entry_point_it != entry_points.end()) {
+                mapped_entry_points.insert(*entry_point_it);
+                continue;
+            }
+
+            const auto& consumer = consumer_input.get_expr();
+            const auto loop_ids = consumer->get_loop_ids();
+            if (!is_loop_id_found(loop_ids, loop_id)) {
+                outside_consumers.insert(consumer);
+            }
+        }
+
+        // Remove entry points which are mapped
+        auto last_point = entry_points.end();
+        for (const auto& mapped_entry_point : mapped_entry_points) {
+            last_point = std::remove(entry_points.begin(), last_point, mapped_entry_point);
+        }
+        entry_points.resize(entry_points.size() - mapped_entry_points.size());
+
+        // Leave exit point if there are consumers outside after fusion
+        if (!outside_consumers.empty()) {
+            new_exit_points.push_back(exit_point);
+        }
+    }
+
+    exit_points = new_exit_points;
+}
+
+void LinearIR::LoopManager::update_loop_port(size_t loop_id, const ExpressionPort& actual_port, const std::vector<ExpressionPort>& target_ports,
+                                             bool is_entry) {
+    const auto& loop_info = get_loop_info(loop_id);
+    auto& ports = is_entry ? loop_info->entry_points : loop_info->exit_points;
+    auto port_it = std::find_if(ports.begin(), ports.end(),
+                                [&actual_port](const LoopPort& point) { return *point.expr_port.get() == actual_port; });
+    // In some cases actual ExpressionPort may not be LoopPort. We shouldn't throw exception here since ExpressionPort is not strong condition as LoopPort
+    // For example, not all inner loop ports are ports of outer loops
+    if (port_it == ports.end())
+        return;
+
+    // to save other parameters except expression port
+    std::vector<LoopPort> target_loop_ports(target_ports.size(), *port_it);
+    std::transform(target_loop_ports.begin(), target_loop_ports.end(), target_ports.begin(), target_loop_ports.begin(),
+                   [](const LoopPort& loop_port, const ExpressionPort& expr_port) {
+                       LoopPort copy = loop_port;  // to save loop port parameters
```

Comment:
Minor: we can pass the first arg by value, so it would be copied automatically. This will save you a couple code lines.